### PR TITLE
Jcolladokuri/variables in logs res path

### LINKED ISSUE #43618: Azure Monitor DS: Logs resource picker does not have support for variables.
**What happened**:
On a dashboard with variables if a panel is created with Azure Monitor datasource and Logs as the service, when a variable is selected as the resource, Grafana crashes and throws the error "PathNotFound": 
image

After some inspection, it seems as though it is because the resource path is not being properly populated, it is rather just injecting the resource name, not a full path: 
image

**What you expected to happen**:

For the same resource selected above, except this time not through variables, this is what the user should see: 
image

and what the resource path should look like
image

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a dashboard with variables using the Azure Monitor data source. 
2. Add a panel with the data source as Azure Monitor and the service as Logs
3. Select any of the variables in the dashboard as the resource path
4. Add a simple query for grafana to run.

**Anything else we need to know?**:

**Environment**: 
- Grafana version: 8.4.0-pre
- Data source type & version: Azure Monitor
- OS Grafana is installed on: Windows
- User OS & Browser: Windows & Edge
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by jcolladokuri**
**What this PR does / why we need it**: This PR adds support for variables when selected for the resource path. Previously, when a variable was selected, the query would break as it did not complete the path properly. 

**Which issue(s) this PR fixes**: not being able to select variables for the resource path when running queries.

image

Fixes #43618

**Special notes for your reviewer**:

### CODE REVIEW

**Review by andresmgot — COMMENTED** [review]
Hi @jcolladokuri,

In order for us to evaluate this PR, please create an issue first with detailed instructions to about how to reproduce the issue. Once we have verified and evaluated the issue we can discuss a solution.

Also, as a general comment, when submitting code:
 - Add new tests that cover the bug fix or the new feature. Make sure you don't introduce regressions.
 - Avoid the usage of the type `any`.

**Comment by jcolladokuri** [comment]
Hi @andresmgot 

Thank you for the feedback and sorry for the delay, I was off for the Holidays! I have gone ahead and created a github issue see: https://github.com/grafana/grafana/issues/43618 

I am working on adding test cases.

Regarding the use of the type `any` I had to cast from `VariableModel` to `any` because the object `VariableModel` does not have properties declared that are available during runtime, some of which I needed to use for this fix. See screenshots below: 


This is what is available during runtime: 
image

This is what is declared on the object: 
image

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts [hunk 1]
```diff
@@ -108,15 +108,57 @@ export default class AzureLogAnalyticsDatasource extends DataSourceWithBackend<
...
+    // as avilable properties in the VariableModel object
+    const resourceVar = allVars.find((var_: any) => var_.id === item.resource?.substring(1));
+    const resourceVarRawQuery = (resourceVar as any)?.query?.grafanaTemplateVariableFn.rawQuery;
+    var subscription = (allVars.find((var_: any) =>
+      var_.query.grafanaTemplateVariableFn.rawQuery.includes('Subscriptions(')
+    ) as any)?.current.value;
+
+    if (resourceVarRawQuery?.includes('Subscriptions(')) {
+      resPath = '/subscriptions/' + resource;
+      resource = resPath;
```
Comment (shuotli) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts:140]: `resPath` doesn't seem necessary. We can simply do `resource = '/subscriptions/' + resource;`
Same with below.

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.test.ts [hunk 2]
```diff
@@ -328,4 +330,107 @@ describe('AzureLogAnalyticsDatasource', () => {
...
+        },
+      };
+
+      const parsedResponse = laDatasource.applyTemplateVariables(query, scopedVars, true);
+      expect(
+        parsedResponse?.azureLogAnalytics?.resource ===
+          '/susbcriptions/testSubID/resourceGroups/testRG/providers/Microsoft.Insights/testNS/testRes'
+      );
+    });
+  });
```
Comment (shuotli) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.test.ts:435]: How about the case when the variable query is invalid?

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts (hunk 1 above)
Comment (jcolladokuri) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts]: done

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.test.ts (hunk 2 above)
Comment (jcolladokuri) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.test.ts:457]: Added test for that

**Pull request closed without merging**